weaver/store/mongodb.py

```diff
@@ -477,6 +526,9 @@ def find_jobs(self,
         elif status:
             search_filters["status"] = status
 
+        if notification_email is not None:
+            search_filters["notification_email"] = notification_email
```

Comment:
I guess that we should search the encrypted mail and not the real mail?

Reply:
Done [here](https://github.com/crim-ca/weaver/pull/46/files/06a87735547afbf6a390868999e7d8f45557c4bf#diff-983083985a48e53e4a69189e6dbe2c53R207) and validated by [this test](https://github.com/crim-ca/weaver/pull/46/files/06a87735547afbf6a390868999e7d8f45557c4bf#diff-19c25203fab87905fceae3782f9a36ecR308)

Reply:
@tomLandry 
To filter by email, you have to specify the literal email to match against their hashed version in the db.
You can't search with the hashed version. 
Unless you have the secret & salt values to know how to pass from one-another, you can't find which hash corresponds to which email literal value, so identifies masking is preserved.

Furthermore, email values are not returned in responses. 
So you can only go by trial-error of literal email values to attempt matching jobs, and if you happen to match jobs this way, then you already have the email to create the request in the first place.